### PR TITLE
Feature: Added location (Geo-coordinates) validation and removed redundant unit tests.

### PR DESCRIPTION
**Pull request by llwasampijja**
I welcome all feedback and advice.
Thank you comrades, Jah Bless, Selah!
@WalujoEmmanuel @ringtho @KapsonLabs @MukuruH @bakersen @Kyeza @nadralia @Kwandera @JammyKalungi @reifred @ernesthenry @jennizalwango @jkMugabe @richien @Araali1 @Mastula77 @Genza999 @Wabuluka @vicentbaiga @noahkalyesubula @LynetteMK @joekirumira @bashirdeveloper @lizz24 @kalsmic @erioc @mwes23 @NANKYAPHIONA @HarunaIbnYusuf @k7ko @v1b3m @codephillip @SimiCode @kdenno, @CandySusan @bekeplar @masete @Anguandia

### CODE REVIEW

**Comment by kalsmic** [comment]
**Request body**
```
{
  "firstname": "Arthur",
  "lastname": "Kalule",
  "othernames": "G",
  "email": "kalulearthur@gmail.com",
  "phonenumber": "0772019937",
  "username": "kalsmic",
  "password": "Password123"
}
```
**Response:**
```
{
    "message": "Entered an empty field or an invalid email address, phonenumber or password",
    "status": 400
}
```
**Note**
* User cannot tell exactly what is wrong with the input.
* Other names can be blank because not everyone has them.
* Use a more descriptive generic message if you still plan to use one i.e
    * How long should the password be and what is the expected combination.



**Request Body**
```
{
  "firstname": "234",
  "lastname": "12",
  "othernames": "242",
  "email": "arthurkalule@gmail.com",
  "phonenumber": "0772019938",
  "username": "123",
  "password": "Password123@"
}
```
**Response**
```
{
    "data": [
        {
            "email": "arthurkalule@gmail.com",
            "firstname": "234",
            "is_admin": false,
            "lastname": "12",
            "othernames": "242",
            "phonenumber": "0772019937",
            "registered_on": "Tue, 08 Jan 2019 05:11:07 GMT",
            "user_id": 4,
            "username": "123"
        }
    ],
    "message": "User account created successifully",
    "status": 201
}
```
**Note**
* Name fields should not contain numbers
* Username can be a combination of numbers and strings but  not numbers only
* Email validation is okay
* Correct the typo in your success message

File: README.md [hunk 1]
```diff
@@ -78,21 +86,36 @@ To run this application, while in the root folder of the project via the Termina
...
+#### Endpoints included
 |Endpoint|Endpoint Purpose|Allowed HTTP Method|Requirements|
 |---|---|---|---|
-| /red-flags  | Get red-flags  |GET  | None |
-| /red-flags/redflag_id | Get red-flag of the specified id  |GET  |None  |
-| /red-flags | Create a red-flag  |POST  |•	created_by (String), location (String), status (String), images ([String, String]), videos ([String, String]), comment (String) |
-| /red-flags/redflag_id/location  | Edit location of a red-flag of specicif id  |PATCH  |•	location(String) |
-| /red-flags/redflag_id  | Delete a red-flag of specific id  |DELETE  |None  |
+| [/](https://ireporter-challenge-two.herokuapp.com/api/v1)  | Index Endpointr  |GET  | None  |
+| [/auth/register](https://ireporter-challenge-two.herokuapp.com/api/v1/auth/register)  | Register/signup a user  |POST  | firstname (String), lastname (String), othernames (String), email (String), phonenumber (String), username (String), password (String)  |
```
Comment (kalsmic) [inline README.md:93]: Endpoint auth/register does not match auth/signup

File: app/controllers/incident_controller.py [hunk 2]
```diff
@@ -153,7 +198,9 @@ def response_unaccepted(self, word):
             "message": message
         }), content_type="application/json", status=status_code)
 
-    def response_sumission_success(self, return_data, keyword):
+    @staticmethod
+    def response_sumission_success(return_data, keyword):
```
Comment (kalsmic) [inline app/controllers/incident_controller.py:202]: Correct typo in the function name. i.e from "response_sumission_success" to "response_submission _success"

File: app/models/incident_model.py [hunk 3]
```diff
@@ -44,68 +48,85 @@ def __init__(self):
...
         return None
 
-    def delete(self, incidednts_list, incident_id, username):
+    @staticmethod
+    def delete(incidednts_list, incident_id, username):
+        """method with the logic to delete an incident in the incidents list"""
         for incident in incidednts_list:
-            if incident.get("incident_id") == incident_id and username == incident.get("created_by"):
+            if incident.get("incident_id") == incident_id \
+            and username == incident.get("created_by"):
```
Comment (kalsmic) [inline app/models/incident_model.py:99]: You should also check for the status of the incident before deleting it.

> The user can only​ ​ edit or delete a ​ red-flag​ / ​ intervention​ record when the record’s ​ status is yet to be marked as either ​ under investigation, rejected, or resolved​ .

**Review by kalsmic — COMMENTED** [review]
Your error messages are so generic and do not guide the user on what is exactly wrong with the input.
Also Correct typos in your code in the function and variable names and also in the response messages.

**Comment by llwasampijja** [comment]
> **Request body**
> 
> ```
> {
>   "firstname": "Arthur",
>   "lastname": "Kalule",
>   "othernames": "G",
>   "email": "kalulearthur@gmail.com",
>   "phonenumber": "0772019937",
>   "username": "kalsmic",
>   "password": "Password123"
> }
> ```
> **Response:**
> 
> ```
> {
>     "message": "Entered an empty field or an invalid email address, phonenumber or password",
>     "status": 400
> }
> ```
> **Note**
> 
> * User cannot tell exactly what is wrong with the input.
> * Other names can be blank because not everyone has them.
> * Use a more descriptive generic message if you still plan to use one i.e
>   
>   * How long should the password be and what is the expected combination.
> 
> **Request Body**
> 
> ```
> {
>   "firstname": "234",
>   "lastname": "12",
>   "othernames": "242",
>   "email": "arthurkalule@gmail.com",
>   "phonenumber": "0772019938",
>   "username": "123",
>   "password": "Password123@"
> }
> ```
> **Response**
> 
> ```
> {
>     "data": [
>         {
>             "email": "arthurkalule@gmail.com",
>             "firstname": "234",
>             "is_admin": false,
>             "lastname": "12",
>             "othernames": "242",
>             "phonenumber": "0772019937",
>             "registered_on": "Tue, 08 Jan 2019 05:11:07 GMT",
>             "user_id": 4,
>             "username": "123"
>         }
>     ],
>     "message": "User account created successifully",
>     "status": 201
> }
> ```
> **Note**
> 
> * Name fields should not contain numbers
> * Username can be a combination of numbers and strings but  not numbers only
> * Email validation is okay
> * Correct the typo in your success message

Thanks, @kalsmic, let me work on that, Bless, Selah!

File: app/models/incident_model.py (hunk 3 above)
Comment (llwasampijja) [reply app/models/incident_model.py:99]: Am on it!

> You should also check for the status of the incident before deleting it.
> 
> > The user can only​ ​ edit or delete a ​ red-flag​ / ​ intervention​ record when the record’s ​ status is yet to be marked as either ​ under investigation, rejected, or resolved​ .

Am on it, I thought I had already implemented this. Let me check to see why this is not working as required.
Thanks again!

File: README.md (hunk 1 above)
Comment (llwasampijja) [reply README.md:93]: @kalsmic I don't get the "Endpoint auth/register does not match auth/signup" comment.  Are you referring to the description of the endpoint where I stated "Register/signup a user"? 
could you please shed more light on that?

Comment (kalsmic) [reply README.md:93]: Check on Line 93 in your readme

Comment (kalsmic) [reply README.md:93]: Sorry My bad, you do not need to change it, i read it from the source code, it refers to the functionality not the endpoint.

**Pull request closed without merging**